drivers/SmartThings/deepsmart/src/deepsmart/dp2knx.lua

```diff
@@ -0,0 +1,177 @@
+local log = require('log')
+local json = require('st.json')
+local config = require('config')
+
+local Dp2Knx = {}
+Dp2Knx.__index = Dp2Knx
+Dp2Knx.pids = {}
+
+--dp2knx
+----pid
+----type
+----dev_n
+------dpid
+----dpid:addrType
+---------------
+-- init default pid to devtype
+---------------
+function Dp2Knx.init()
+  -- devs is array
+  -- for pid n0ykidbbn3mqvxbx, it will split to 3 switchs to samsung
+  -- so idx(1,2,3) is used for sub devices
+  local pid2 = {pid='cekfhkz5',type=config.ENUM.AC, devs={[1]={[1]=0,[4]=1,[5]=2,[2]=3,[3]=4,[101]=5}}}
+  local pid4 = {pid='drrearrq',type=config.ENUM.HEATER, devs={[1]={[1]=0,[24]=2,[16]=3}}}
+  local pid53 = {pid='w35ineur',type=config.ENUM.NEWFAN, devs={[1]={[1]=0,[12]=1}}}
+  Dp2Knx.pids[pid2.pid] = pid2
+  Dp2Knx.pids[pid4.pid] = pid4
+  Dp2Knx.pids[pid53.pid] = pid53
+  return 0
+end
```

Comment:
```suggestion
-- for pid n0ykidbbn3mqvxbx, it will split to 3 switchs to samsung
-- so idx(1,2,3) is used for sub devices
Dp2Knx.pids = {
  ['cekfhkz5'] = {pid='cekfhkz5',type=config.ENUM.AC, devs={[1]={[1]=0,[4]=1,[5]=2,[2]=3,[3]=4,[101]=5}}},
  ['drrearrq'] = {pid='drrearrq',type=config.ENUM.HEATER, devs={[1]={[1]=0,[24]=2,[16]=3}}},
  ['w35ineur'] = {pid='w35ineur',type=config.ENUM.NEWFAN, devs={[1]={[1]=0,[12]=1}}}
}
```
The init function for dp2kn and dpenum can both be removed

Reply:
delete init function and init in default constructor